app.py

```diff
@@ -185,17 +186,37 @@ def set_frame_range(self, engine, in_frame, out_frame):
 
         elif engine == "tk-nuke":
             import nuke
-
+            root = nuke.root()
+            lock_range = root["lock_range"]
             # unlock
-            locked = nuke.root()["lock_range"].value()
+            locked = lock_range.value()
+            change_msg = 'from %s-%s to %s-%s' % (prev_in, prev_out, in_frame, out_frame)
             if locked:
-                nuke.root()["lock_range"].setValue(False)
+                ret = nuke.ask("Frame range is currently locked,\n"
+                               "OK to update frame range %s ?" % change_msg)
```

Comment:
Talking from a Tank perspective, `Do you want to update frame range %s` would be more tanky than `OK`, as they normally don't use such phrases.